backend/src/backend/main.py

```diff
@@ -53,27 +60,39 @@ async def test(timestamp: int):
     return {"requested_on": requested_on, "received_on": received_on}
 
 
+@database.transaction()
 @api.post("/books/add/", status_code=201)
-async def books_add(book: SchemaBook):
-    await Book.objects.create(
-        id=book.id,
-        counter=book.counter,
-        period=book.period,
-        article_count=book.article_count,
-        media_count=book.media_count,
-        size=book.size,
-        url=book.url,
-        zimcheck=book.zimcheck,
+async def add_book(book_payload: BookAddSchema):
+    """API endpoint to receive Book addition requests and add to database"""
+
+    book = await Book.objects.create(
+        id=book_payload.id,
+        counter=book_payload.counter,
+        period=book_payload.period,
+        article_count=book_payload.article_count,
+        media_count=book_payload.media_count,
+        size=book_payload.size,
+        url=book_payload.url,
+        zimcheck=book_payload.zimcheck,
     )
 
-    for key, value in book.metadata.items():
+    for metadata_name, value in book_payload.metadata.items():
+        kind = (
+            KIND_TEXT
+            if not metadata_name.startswith("Illustration_")
+            else KIND_ILLUSTRATION
+        )
         await BookMetadata.objects.create(
             book=book.id,
-            name=key,
+            name=metadata_name,
             value=value,
```

Comment:
If you look at the model, you'll see that we store the content into different fields for binary/illustration. You want to save the decoded base64 into `bin_value`. This allows us to run text-only queries on text metadata without hassle.